tools/deployJournals.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+const path_     = require('path');
+const fs_       = require('fs');
+const path = path_.posix;
+
+require('../src/foam_node.js');
+
+var [argv, X, flags] = require('./processArgs.js')(
+  '',
+  { version: '', license: '', pom: 'pom' },
+  { debug: true, java: false, web: true }
+);
+
+foam.require(X.pom, false, true);
+
+const outPath = 'runtime/journals';
+
+if ( ! fs_.existsSync(outPath) ) {
+    fs_.mkdirSync(outPath, { recursive: true });
+}
+
+async function findJournals({ jrls, srcPath, jrlName }) {
+    const walker = foam.util.filesystem.FileWalker.create();
+    walker.files.sub((_1, _2, info) => {
+        for ( const fileInfo of info.files ) {
+            if ( foam.poms[fileInfo.path] != undefined ) continue;
+            const extName = path_.extname(fileInfo.name);
+            if ( extName !== '' && extName !== '.jrl' ) continue;
+            if ( jrlName && fileInfo.name != jrlName ) continue;
+            const fullPath = fileInfo.fullPath;
+            const baseName = path_.basename(fileInfo.name, extName);
+            if ( ! jrls[baseName] ) jrls[baseName] = '';
+            jrls[baseName] += fs_.readFileSync(fullPath).toString();
+            if ( jrlName && fileInfo.name == jrlName ) return;
+        }
+    })
+    await walker.walk(srcPath);
+}
+
+
+ const main = async function() {
+
+ foam.require(X.pom, false, true);
+ const jrls = [];
+ asyncForEach(Object.keys(foam.poms), async(p) => {
+   if ( foam.poms[p].journals ) {
+     foam.poms[p].journals.forEach(async (j) => {
+      await findJournals({jrls, srcPath: p, jrlName: j});
```

Comment:
@kgrgreer if I understood correclty, we want `.journals` of the pom to be a list of explicit paths relative to the POM's location rather than just names, is that right?

Reply:
if a pom requires users journals for example but there are multiple users under that directory, assuming that projects are organized properly, we can say that we need all users jrls, so wouldn't it be better to just specify users rather than list every single path for each? Although it would speed up the deployment process but its is already fairly fast so i dont know if it needs optimization

Reply:
I think the rationale is to be consistent with `files` - by specifying the files we can specify the order they're loaded in, and that can be also true for journals (which may allow a package to override just a few properties of a journal entry that already exists in the project)

when journals are auto-loaded (i.e. if you didn't specify `journals:`) the order of directories is platform-dependent (I found an [interesting github issue](https://github.com/nodejs/node/issues/3232) describing this). We could also sort it alphabetically so that `a/b/users.jrl` is loaded before `a/c/users.jrl`, but that's an arbitrary ordering just to ensure consistency across platforms